### PR TITLE
Eng 1322 add UI for showing user what service is used for metadata storage

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
Display the storage layer type.

Adding the integration name will be a bigger effort and we should make a new task for that if we want to pursue that: https://aqueducthq.slack.com/archives/C01U6RREC7M/p1664834954509439

## Related issue number (if any)
Eng 1322

## Loom demo (if any)
image

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [N/A] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Comment by agiron123** [comment]
Nice work. Curious why this is a dropdown. Can the user change the service that is being used from this UI?

If they can't, then this should probably be text.

**Comment by eunice-chan** [comment]
@agiron123 Currently, we do not allow the user to change the service used by the storage layer but this is planned feature and has a task on linear. Since it is not currently an implemented feature, should I comment out the dropdown and replace it with a text for now?

File: src/ui/common/src/components/workflows/storageSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { RootState } from '../../stores/store';
+import { StorageType, StorageTypeNames } from '../../utils/storage';
+
+export const StorageSelector: React.FC = () => {
+  const workflow = useSelector((state: RootState) => state.workflowReducer);
+  const dag = workflow.selectedDag;
+  let selected = 'file';
+  if (dag) {
+    selected = dag.storage_config.type;
+    console.log('selected dag', dag.storage_config.type);
```
Comment (saurav-c) [inline src/ui/common/src/components/workflows/storageSelector.tsx:18]: nit: remove logs

**Review by saurav-c — APPROVED** [review]
LGTM, so for now we will just show the type of the storage instead of the config or integration name?

**Comment by agiron123** [comment]
> @agiron123 Currently, we do not allow the user to change the service used by the storage layer but this is planned feature and has a task on linear. Since it is not currently an implemented feature, should I comment out the dropdown and replace it with a text for now?

@eunice-chan this is fine for now. Let's go with this design. I have a task to clean up the workflow header and make it more responsive, so I might look into using a text instead then. But for now, I think what you have should be fine.

File: src/ui/common/src/components/workflows/storageSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  if (dag) {
+    selected = dag.storage_config.type;
+  }
+
+  const getMenuItems = () => {
+    return (
+      Object.values(StorageType) as Array<
+        typeof StorageType[keyof typeof StorageType]
+      >
+    ).map((r, _) => {
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/storageSelector.tsx:25]: I think you can just omit the second parameter of .map here.

Try doing:
```typescript
.map((r) => ...
```

Comment (agiron123) [reply src/ui/common/src/components/workflows/storageSelector.tsx:25]: Could you also use a more descriptive name here please? Something like storageType should suffice.

File: src/ui/common/src/components/workflows/storageSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  const menuItems = getMenuItems();
+
+  return (
+    <Box sx={{ display: 'flex', alignItems: 'center' }}>
+      Storage Type:{' '}
+      <FormControl disabled sx={{ m: 1, minWidth: 120 }} size="small">
+        <Select
+          sx={{ maxHeight: 50 }}
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/storageSelector.tsx:47]: Can you use a unit of 8 here like 48 or 56 please?

**Review by agiron123 — APPROVED** [review]

File: src/ui/common/src/components/workflows/storageSelector.tsx (hunk 3 above)
Comment (eunice-chan) [reply src/ui/common/src/components/workflows/storageSelector.tsx:47]: I used the same structure & styling as the VersionSelector (except I added `m: 1` for this component.) I've updated both to be units of 8 (maxHeight: 48).

It doesn't seem to have much of a change to the UI:
image

**Pull request merged**